paderbox/array/sparse.py

```diff
@@ -0,0 +1,1082 @@
+        >>> a[:5] = np.ones(3)
+        Traceback (most recent call last):
+         ...
+        ValueError: Could not assign array with shape (3,) to SparseArray slice with size 5
+        >>> a[:5] = np.ones((1, 5))
+        Traceback (most recent call last):
+         ...
+        ValueError: Shape mismatch: SparseArray has shape (10,), but assigned array has shape (1, 5).
+        >>> a[:100] = np.ones(100)
+        Traceback (most recent call last):
+         ...
+        ValueError: Could not assign array with shape (100,) to SparseArray slice with size 10
+        >>> a = SparseArray((5, 10))
+        >>> a[:4, :] = 1
+        Traceback (most recent call last):
+          ...
+        IndexError: Cannot partially set values in the leading dimensions ([slice(None, 4, None)] at dimension 0)
+        >>> a[0, :] = 1
+        Traceback (most recent call last):
+          ...
+        IndexError: SparseArray.__setitem__ only supports slices for indexing, not [0, slice(None, None, None)]
+        >>> a[:, 0] = 1
+        Traceback (most recent call last):
+          ...
+        IndexError: SparseArray.__setitem__ only supports slices for indexing, not [slice(None, None, None), 0]
+        """
+        item = _normalize_item(item, self.shape)
+
+        length = len(item)
+        for idx, i in enumerate(item):
+            # All items must be slices because we can't set scalar values
+            if not isinstance(i, slice):
+                raise IndexError(
+                    f'{self.__class__.__name__}.__setitem__ only supports '
+                    f'slices for indexing, not {item!r}'
+                )
+
+            # We don't support steps
+            if i.step is not None:
+                raise IndexError(f'Step is not supported.')
+
+            # All slices except for the last one must not have a start and stop
+            # value set. The leading dimensions of all arrays must be equal.
+            if idx < length - 1 and (i.start is not None or i.stop is not None):
+                raise IndexError(
+                    f'Cannot partially set values in the leading dimensions '
+                    f'({item[:-1]!r} at dimension {idx})'
+                )
+
+        # We only need to handle the last element in item, all others are
+        # slice(None)
+        item = item[-1]
+        start, stop = cy_parse_item(item, self.shape)
+        length = stop - start
+
+        if np.isscalar(value):
+            # Infer the dtype from value if the SparseArray doesn't have a
+            # dtype yet
+            value = self._new_full(
+                shape=self.shape[:-1] + (length,), fill_value=value,
+                dtype=self.dtype if self.dtype else _dtype_from_value(value)
+            )
+
+        if isinstance(value, np.ndarray) or isinstance(value, torch.Tensor):
+            # The last dimension has to match the length defined by the indexing
+            if value.shape[-1] != length:
```

Comment:
Add an ndim check.

Reply:
Is implicitly done in the if statement below

Reply:
ok